libstorage/drivers/storage/cinder/storage/cinder_storage.go

```diff
@@ -672,7 +675,26 @@ func (d *driver) VolumeAttach(
 			fields, "error waiting for volume to attach", err)
 	}
 
-	return volume, volumeAttach.Device, nil
+	return volume, resolveDeviceName(volumeID, volumeAttach.Device), nil
+}
+
+func getDeviceLink(
+	volumeID string) string {
+
+	return fmt.Sprintf("/dev/disk/by-id/virtio-%s", volumeID[:20])
```

Comment:
The bit about the executor aside, is this universal across OpenStack/Cinder? I think one of the reasons things are the way they are now is because not everything is using virtio, correct? What if this was on a Xen hypervisor? Would this path be the same?

I recognize things are broken right now, and we should target the most common usecase, which is probably KVM with virtio. But if that's not universal, this needs to be a configurable behavior.

Reply:
`virtio-%s` this won't work in all cases, as for example in my case, devices are named as:

```
/dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_2afa0091-d1c5-46e2-9
/dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_426c308d-a355-4cca-a
/dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_4dfda350-9cd9-4901-8
/dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_52645ecc-2ac0-4534-a
/dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_5569888b-5d88-4a57-9
/dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_698a3ded-990a-420b-8
```

(Openstack from OVH)

Reply:
ICYMI, for OVH/Openstack I hacked a variant and made the plugin available here: https://hub.docker.com/r/hervenicol/rexray.

Reply:
...it appears the hack is not enough, we still have a lot of issues: wrong volume mounted, volumes created although already existing, unable to mount volumes...
So I decided to go to a simpler, cinder-dedicated plugin: https://github.com/hervenicol/docker-plugin-cinder

It means a lot less code, easier to understand / debug / evolve.
It seems rexray was overkill for my use case. And on top of that, too complex for me to maintain / fix it by myself.